app/uk/gov/hmrc/economiccrimelevyregistration/connectors/EclRegistrationConnector.scala

```diff
@@ -80,6 +82,12 @@ class EclRegistrationConnector @Inject() (appConfig: AppConfig, httpClient: Http
         case Left(e)             => throw e
       }
 
-  def submitRegistration(internalId: String)(implicit hc: HeaderCarrier): Future[CreateEclSubscriptionResponse] =
-    httpClient.POSTEmpty[CreateEclSubscriptionResponse](s"$eclRegistrationUrl/submit-registration/$internalId")
+  def submitRegistration(internalId: String, entityType: Option[EntityType])(implicit
+    hc: HeaderCarrier
+  ): Future[CreateEclSubscriptionResponse] =
+    entityType match {
+      case Some(Other) => Future.successful(CreateEclSubscriptionResponse(Instant.now, ""))
+      case _           =>
+        httpClient.POSTEmpty[CreateEclSubscriptionResponse](s"$eclRegistrationUrl/submit-registration/$internalId")
+    }
```

Comment:
I think we need to rethink this approach - we shouldn't be constructing an empty response like this and in fact I'd go as far as saying the backend should probably have a separate API all together when it comes to sending the Other data PDF to DMS.

I would avoid changing anything about this method, and instead just make the controller not expect any call to be made for now, until the backend API is implemented

Reply:
You are still submitting a registration for all entities. The submission is different for Other entities that for other types.

The check for what to do depending on entity type should be made on the connector. It will call a different API but for now it will do nothing for Other entities.

Reply:
I agree the backend will still be called eventually

But I would not advise using the same connector method to calling different endpoints based on some logic, the idea of the connectors is that you define a method for each call to each endpoint. Any logic for calling the connector should be applied in a controller/service layer - so we don't need to do anything to the connector at the moment

This is why i've suggested logic in the controller that only makes the call if it's a normal entity for now, because it's the optionality of the Response we care about. The other submission will not return this response and infact will not return anything useful other than a 2xx - so this logic should definitely be in the controller

Reply:
To elaborate further on this - I think once we do the backend work, we should go a step further and de-couple the submission/validation logic for 'other' registrations from 'normal' registrations. So have a separate endpoint all together that calls a different validate method instead of relying on the `Either[...]` type that we originally introduced.

This would mean that a new connector method would be added on the frontend to return `HttpResponse` and when it is called as part of the for comprehension in the controller, it is done like so `_ <- eclRegistrationConnector.submitOtherRegistration` - since we don't care about the result, only that it was 2xx

Reply:
Moved logic out of connector.